src/methods/sep10Auth/sign.ts

```diff
@@ -0,0 +1,32 @@
+import { Keypair, Transaction, xdr } from "stellar-sdk";
+import { log } from "helpers/log";
+
+export const sign = ({
+  challengeTransaction,
+  networkPassphrase,
+  secretKey,
+}: {
+  challengeTransaction: any;
+  networkPassphrase: string;
+  secretKey: string;
+}) => {
+  log.instruction({
+    title:
+      "We've received a challenge transaction from the server that we need the sending anchor to sign with their Stellar private key.",
+  });
+
+  const envelope = xdr.TransactionEnvelope.fromXDR(
+    challengeTransaction,
+    "base64",
+  );
+  const transaction = new Transaction(envelope, networkPassphrase);
```

Comment:
We should validate the challenge transaction according to the SEP-10 specification. We can use the JS SDK's [readChallengeTx()](https://stellar.github.io/js-stellar-sdk/Utils.html#.readChallengeTx) for that.

Reply:
Fixed in https://github.com/stellar/stellar-demo-wallet/pull/125